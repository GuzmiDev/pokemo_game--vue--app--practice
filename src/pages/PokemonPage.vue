<template>
  <h1>¿Quién es este pokémon?</h1>
  <PokemonPicture :pokemonId="386" :showPokemon="true" />
  <PokemonOptions :pokemons="pokmeonArr" />
</template>

<script>
import PokemonPicture from "@/components/PokemonPicture.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";

import getPokemonOptions from "@/helpers/getPokemonOptions";

export default {
  name: "PokemonPage",
  components: {
    PokemonPicture,
    PokemonOptions,
  },
  data() {
    return {
      pokmeonArr: [],
    };
  },
  methods: {
    async mixPokemonArray() {
      this.pokmeonArr = await getPokemonOptions();
    },
  },
  mounted() {
    this.mixPokemonArray();
  },
};
</script>
