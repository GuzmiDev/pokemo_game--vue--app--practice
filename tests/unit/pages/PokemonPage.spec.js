import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage";
import { pokemons } from "../mocks/pokoemons.mock";

describe("Pruebas en PokemonPage", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("debe mostrarse correctamente", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de llamar mixPokemonArray al montar", () => {
    const mixPokemonArraySpy = jest.spyOn(
      PokemonPage.methods,
      "mixPokemonArray"
    );

    shallowMount(PokemonPage);

    expect(mixPokemonArraySpy).toHaveBeenCalled();
  });

  test("debe de mostrarse correctamente cuando cargan los pokemons", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokmeonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: null,
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de mostrar los componentes de PokemonPicture y PokemonOptions", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokmeonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: null,
        };
      },
    });

    const picture = wrapper.find("pokemon-picture-stub");
    const options = wrapper.find("pokemon-options-stub");

    expect(picture.exists()).toBeTruthy();
    expect(options.exists()).toBeTruthy();

    expect(picture.attributes("pokemonid")).toBe(pokemons[0].id.toString());
    expect(options.attributes("pokemons")).toBeTruthy();
  });

  test("pruebas con checkAnswer", async () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokmeonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: null,
        };
      },
    });

    await wrapper.vm.checkAnswer(1);

    expect(wrapper.find("h2").exists()).toBeTruthy();
    expect(wrapper.vm.showPokemon).toBeTruthy();
    expect(wrapper.find("h2").text()).toBe(`Correcto, ${pokemons[0].name}`);

    await wrapper.vm.checkAnswer(4);
    expect(wrapper.find("h2").text()).toBe(`Oops, era ${pokemons[0].name}`);
  });
});
