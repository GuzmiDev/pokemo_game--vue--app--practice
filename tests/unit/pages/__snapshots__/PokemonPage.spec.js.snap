// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PokemonPage debe de mostrarse correctamente cuando cargan los pokemons 1`] = `
<div>
  <h1>¿Quién es este pokémon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
</div>
<!--v-if-->
`;

exports[`Pruebas en PokemonPage debe mostrarse correctamente 1`] = `
<h1>Espere por favor...</h1>
<!--v-if-->
`;
